spec/controllers/test_results_controller_spec.rb

```diff
@@ -0,0 +1,189 @@
+require "rails_helper"
+
+RSpec.describe User::TestResultsController, type: :controller do
+
+  around do |example|
+    I18n.with_locale(:en) { example.run }
+  end
+
+  let!(:user) do
+    User.create!(
+      name: "User Chính",
+      email: "main.user@example.com",
+      password: "password123",
+      password_confirmation: "password123",
+      birthday: "2000-01-15",
+      gender: 0
+    )
+  end
+
+  let!(:another_user) do
+    User.create!(
+      name: "User Khác",
+      email: "another.user@example.com",
+      password: "password123",
+      password_confirmation: "password123",
+      birthday: "1998-10-10",
+      gender: 1
+    )
+  end
+
+  let!(:test_object) do
+    Test.create!(
+      name: "Bài kiểm tra chương 1",
+      description: "Kiến thức cơ bản về Ruby on Rails.",
+      duration: 1800,
+      max_attempts: 3
+    )
+  end
+
+  let!(:course) do
+    Course.create!(
+      title: "Khóa học Lập trình Ruby on Rails",
+      description: "Đây là phần mô tả chi tiết cho khóa học.",
+      duration: 30,
+      creator: user
+    )
+  end
+
+  let!(:lesson) do
+    Lesson.create!(
+      course: course,
+      title: "Bài 1: Giới thiệu về Models",
+      description: "Tìm hiểu về ActiveRecord.",
+      position: 1,
+      creator: user
+    )
+  end
+
+  let!(:component) do
+    Component.create!(
+      lesson: lesson,
+      component_type: 0,
+      test: test_object,
+      index_in_lesson: 1
+    )
+  end
+
+  let!(:test_result) do
+    TestResult.create!(
+      user: user,
+      component: component,
+      attempt_number: 1,
+      mark: 95,
+      status: 1,
+      submitted: true
+    )
+  end
+
+  let!(:another_test_result) do
+    TestResult.create!(
+      user: another_user,
+      component: component,
+      attempt_number: 1,
+      mark: 88,
+      status: 1,
+      submitted: true
+    )
+  end
+
+  describe "GET #show" do
+    context "when user is logged in and accesses their own result" do
+      before do
+        allow(controller).to receive(:current_user).and_return(user)

```

Comment:
Đưa allow(controller).to receive(:current_user).and_return(user) ra 1 before để dùng chung